Development and unit tests

BelarusVadzim opened the pull request:
Development and unit tests

VadzimZhukouski committed:
development and unit tests

aabrazouski commented on .vscode/settings.json:
Add `.vscode/` folder to gitignore, please

aabrazouski commented on src/components/atoms/Button/Button.tsx:
```diff
@@ -0,0 +1,34 @@
+  glyph,
+  className,
+}) => {
+  let combinedClasses = useMemo(
```
`useMemo` is redundant here. Memoization should be only used in a very big calculations. To handle classnames you can use a package https://www.npmjs.com/package/classnames

aabrazouski commented on src/components/atoms/Checkbox/Checkbox.test.tsx:
```diff
@@ -0,0 +1,18 @@
+    expect(component.asFragment()).toMatchSnapshot();
+  });
+
+  it('when checked and classname specified should render properly ', () => {
```
Add a case to check that `onToggle` have been called and checkbox changed the state.

BelarusVadzim replied:
My checkbox is implemented in such a way that it does not store its own state, but displays the state of the model to which it was bound. It is re-rendered when the state of the application changes. As far as I understand, I can do such a check in E2E tests.

aabrazouski replied:
You still can pass mocked function and check if it was called on change

aabrazouski commented on src/components/atoms/Checkbox/Checkbox.tsx:
```diff
@@ -0,0 +1,29 @@
+  combinedClassName = className ? `${combinedClassName} ${className}` : combinedClassName;
+
+  return (
+    <div className={`material-icons ${combinedClassName}`} onClick={onToggle}>
```
Use `input type='checkbox'` for this, please. Markup should be semantic

aabrazouski commented on src/components/atoms/Textbox/Textbox.tsx:
```diff
@@ -0,0 +1,37 @@
+  const keyDownHandler = (keyboardEventArg: KeyboardEvent<HTMLInputElement>) => {
+    if (EnterKey === keyboardEventArg.key) {
+      if (onReturn) 
+        onReturn(keyboardEventArg.currentTarget.value);
```
Wrap if statements into `{}` please

aabrazouski commented on src/components/atoms/Title/Title.tsx:
```diff
@@ -0,0 +1,8 @@
+import style from './Title.module.scss';
+
+const Title: React.FC<PropsWithChildren> = ({ children }) => {
+  return <span className={style.title}>{children}</span>;
```
For titles use `h1...h6` tags.
Actually, there is not need to have this component

aabrazouski commented on src/components/atoms/Title/Title.test.tsx:
```diff
@@ -0,0 +1,12 @@
+import { render } from '@testing-library/react';
+import Title from './Title';
+
+jest.mock('./Title.module.scss', () => ({ title: 'titleStyle' }));
```
What is purpose of mocking styles?

aabrazouski commented on src/App.tsx:
```diff
@@ -0,0 +1,19 @@
+import React, { useEffect } from 'react';
+
+const App: React.FC = () => {
+  const { isAppInitialized, dispatchAppInitialized } = useAppStateService();
```
What is purpose of this hook?

VadzimZhukouski committed:
package.json fix

VadzimZhukouski committed:
Fix prettier

VadzimZhukouski committed:
fix

VadzimZhukouski committed:
fix title

VadzimZhukouski committed:
fix style mock in tests

VadzimZhukouski committed:
Fix: remove appInitialized flag

VadzimZhukouski committed:
Spinner added

VadzimZhukouski committed:
Remove applicationSlice, useAppStateService
Pending property has been added into the state.
A spinner has been added for pending case.

VadzimZhukouski committed:
Fix tests

aabrazouski commented on src/components/atoms/Checkbox/Checkbox.tsx:
```diff
@@ -0,0 +1,31 @@
+    : combinedClassName;
+
+  return (
+    <div className={`material-icons ${combinedClassName}`} onClick={onToggle}>
```
Wrap input into `label` element, please

aabrazouski replied:
Also handle `onChange` (on input) instead of `onClick`. If you try to navigate the app with keyboard you will see that it works incorrectly.

aabrazouski commented on src/components/atoms/Label/Label.tsx:
```diff
@@ -0,0 +1,33 @@
+    draggable?: boolean;
+  };
+
+const Label: React.FC<LabelProps> = ({
```
Why is it called Label?

aabrazouski commented on src/components/molecules/NoteEditor/NoteEditor.tsx:
```diff
@@ -0,0 +1,32 @@
+
+  const editFinished = (text: string) => {
+    if (text)
+      if (note) dispatchTodoItemEdited({ ...note, text });
```
Please, always wrap code blocks into `{}`

aabrazouski commented on src/components/molecules/Spinner/Spinner.module.scss:
```diff
@@ -0,0 +1,25 @@
+  left: 0;
+  height: 100%;
+  width: 100%;
+  //display: none;
```
Delete it if in not needed, please

VadzimZhukouski committed:
fix after review

VadzimZhukouski committed:
Checkbox and button fix

VadzimZhukouski committed:
fix

aabrazouski commented on src/components/atoms/Checkbox/Checkbox.tsx:
```diff
@@ -0,0 +1,42 @@
+    : combinedClassName;
+
+  /*
+  This approach allows the component to be highlighted when navigating with the keyboard, but not when using the mouse.
```
Didn't get your problem. Highlighting controlled via css not js

aabrazouski commented on src/components/molecules/TodoItem/TodoItem.tsx:
```diff
@@ -0,0 +1,48 @@
+        className={style.checkbox}
+        onToggle={checkBoxToggle}
+      />
+      <TextPlace
```
This text should be inside the label of `Checkbox` component

aabrazouski commented on src/components/atoms/RadioButton/RadioButton.tsx:
```diff
@@ -0,0 +1,40 @@
+  value,
+  className = '',
+}) => {
+  const combinedClassName = `${style.radioButton} ${className}`;
```
use `classnames` package please

aabrazouski commented on src/components/atoms/Button/Button.tsx:
```diff
@@ -0,0 +1,30 @@
+type ButtonProps = PropsWithClassName & {
+  value: string;
+  onClick?: () => void;
+  glyph?: boolean;
```
A very confusing prop name
